config/file_config.go

```diff
@@ -146,6 +143,8 @@ func NewConfig(config, rules string, errorCallback func(error)) (Config, error)
 
 	r.SetDefault("Sampler", "DeterministicSampler")
```

Comment:
Does these need to happen after the config has been loaded? Could they all be done beforehand?

Reply:
They don't need to happen after. I think this bit is just meant to continue the distinction between the general config file and the rules file, since these values are all set in the rules file. @martin308 Is that correct?

Reply:
we need to do them beforehand AFAIK, which is the case here as we don't load the config until line 150

Reply:
Ah, just seen the `SetDefaults` are on the `r` not the `c` object so can't happen any sooner 👍